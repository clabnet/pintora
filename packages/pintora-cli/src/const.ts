export const SVG_MIME_TYPE = 'image/svg+xml'

export const SUPPORTED_MIME_TYPES = ['image/png', 'image/jpeg', SVG_MIME_TYPE]

export const DEFAUT_BGS = {
  light: '#FFFFFF',
  dark: '#282A36',
}
