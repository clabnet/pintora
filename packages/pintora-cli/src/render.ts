import { pintoraStandalone } from '@pintora/standalone'
import { JSDOM } from 'jsdom'
import { implForWrapper } from 'jsdom/lib/jsdom/living/generated/utils'
import { Canvas } from 'canvas'

export const SUPPORTED_MIME_TYPES = [
  'image/png',
  'image/jpeg',
]

export type CLIRenderOptions = {
  code: string
  devicePixelRatio?: number | null
  mimeType?: string
  backgroundColor?: string
  // width?: number
  // height?: number
}

export function render(opts: CLIRenderOptions) {
  const { code, backgroundColor } = opts
  const devicePixelRatio = opts.devicePixelRatio || 2
  const mimeType = opts.mimeType || 'image/png'

  const dom = new JSDOM('<!DOCTYPE html><body></body>')
  const document = dom.window.document
  const container = document.createElement('div')
  container.id = 'pintora-container'

  // setup the env for renderer
  global.window = dom.window as any
  global.document = document
  ;(dom.window as any).devicePixelRatio = devicePixelRatio

  return new Promise<Buffer>((resolve, reject) => {
    pintoraStandalone.renderTo(code, {
      container,
      renderer: 'canvas',
      enhanceGraphicIR(ir) {
        if (backgroundColor && !ir.bgColor) ir.bgColor = backgroundColor
        return ir
      },
      onRender(renderer) {
        setTimeout(() => {
          const buf = getBuf(renderer.getRootElement() as HTMLCanvasElement)
          resolve(buf)
        }, 20)
      },
      onError(e) {
        console.error('onError', e)
      },
    })
  })

  function getBuf(canvas: HTMLCanvasElement) {
    // currently jsdom only support node-canvas,
    // and this is it's not-so-stable method for getting the underlying node-canvas instance
    const wrapper = implForWrapper(canvas)
    const nodeCanvas: Canvas = wrapper._canvas
    const context = nodeCanvas.getContext('2d')
    context.quality = 'best'
    context.patternQuality = 'best'
    console.log('mime type', mimeType)
    const buf = nodeCanvas.toBuffer(mimeType as any)
    return buf
  }
}
