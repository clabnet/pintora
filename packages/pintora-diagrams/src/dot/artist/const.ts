export const DEFAULT_LINE_WIDTH = 1

export const BOLD_LINE_WIDTH = 2

export const DASHED_LINE_DASH = [5, 5]

export const DOTTED_LINE_DASH = [2, 4]
