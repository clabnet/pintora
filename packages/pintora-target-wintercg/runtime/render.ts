import { RenderOptions, IRenderer } from '@pintora/renderer'
import { pintoraStandalone, PintoraConfig, DeepPartial } from '@pintora/standalone'
import './text-metric'
// import { Window } from 'happy-dom'
// import JSDOM from 'jsdom'
import { createHTMLWindow } from 'svgdom'

export const DEFAUT_BGS = {
  light: '#FFFFFF',
  dark: '#282A36',
}

export type RuntimeRenderOptions = {
  /**
   * pintora DSL to render
   */
  code: string
  devicePixelRatio?: number | null
  mimeType?: string
  /**
   * Assign extra background color
   */
  backgroundColor?: string
  pintoraConfig?: DeepPartial<PintoraConfig>
  /**
   * width of the output, height will be calculated according to the diagram content ratio
   */
  width?: number
}

// pintoraStandalone.setConfig({
//   core: {
//     defaultFontFamily: 'sans-serif',
//   },
// })

function renderPrepare(opts: RuntimeRenderOptions) {
  const { code, backgroundColor, pintoraConfig } = opts
  const devicePixelRatio = opts.devicePixelRatio || 2
  // const dom = new JSDOM('<!DOCTYPE html><body></body>')
  // const dom = new Window()

  const window = createHTMLWindow()
  const document = window.document

  const container = document.createElement('div')
  container.id = 'pintora-container'

  const global = globalThis

  // setup the env for renderer
  ;(global as any).document = document
  ;(window as any).devicePixelRatio = devicePixelRatio

  return {
    container,
    pintorRender(renderOpts: Pick<RenderOptions, 'renderer'>) {
      let config = pintoraStandalone.getConfig<PintoraConfig>()
      if (pintoraConfig) {
        config = pintoraStandalone.configApi.gnernateNewConfig(pintoraConfig)
      }

      const containerSize = opts.width ? { width: opts.width } : undefined
      if (opts.width) {
        config = pintoraStandalone.configApi.gnernateNewConfig({ core: { useMaxWidth: true } })
      }

      return new Promise<IRenderer>((resolve, reject) => {
        pintoraStandalone.renderTo(code, {
          container,
          renderer: renderOpts.renderer || 'canvas',
          containerSize,
          enhanceGraphicIR(ir) {
            if (!ir.bgColor) {
              const themeVariables: Partial<PintoraConfig['themeConfig']['themeVariables']> =
                config.themeConfig.themeVariables || {}
              const newBgColor =
                backgroundColor ||
                themeVariables.canvasBackground ||
                (themeVariables.isDark ? DEFAUT_BGS.dark : DEFAUT_BGS.light)
              ir.bgColor = newBgColor
            }
            return ir
          },
          onRender(renderer) {
            resolve(renderer)
          },
          onError(e) {
            console.error('onError', e)
            reject(e)
          },
        })
      })
    },
  }
}

/**
 * Renders the Pintora CLI options to the specified output format.
 * @param opts - The CLIRenderOptions.
 * @returns A promise that resolves to the rendered output.
 */
export function render(opts: RuntimeRenderOptions) {
  const { pintorRender } = renderPrepare(opts)

  function renderToSvg() {
    return new Promise<string>((resolve, reject) => {
      pintorRender({ renderer: 'svg' })
        .then(renderer => {
          const rootElement = renderer.getRootElement() as SVGSVGElement
          rootElement.setAttribute('xmlns', 'http://www.w3.org/2000/svg')
          resolve(rootElement.outerHTML)
        })
        .catch(reject)
    })
  }
  return renderToSvg()
}
