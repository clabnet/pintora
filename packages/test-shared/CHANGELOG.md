# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)


### Features

* **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))





# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)


### Features

* **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))





# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)


### Features

* [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))





# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)


### Bug Fixes

* **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))


### Features

* **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))





# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)


### Features

* Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
