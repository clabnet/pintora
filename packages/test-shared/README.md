# `@pintora/test-shared`

> TODO: description

## Usage

```
const testShared = require('@pintora/test-shared');

// TODO: DEMONSTRATE API
```
