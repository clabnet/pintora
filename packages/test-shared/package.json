{
  "name": "@pintora/test-shared",
  "version": "0.0.0",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  }
}
