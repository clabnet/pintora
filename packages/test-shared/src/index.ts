export { EXAMPLES } from './data/examples'
export { stripStartEmptyLines } from './util'
