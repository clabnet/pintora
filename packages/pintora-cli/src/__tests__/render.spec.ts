import { pintoraStandalone } from '@pintora/standalone'
import { EXAMPLES } from '@pintora/test-shared'
import { SVG_MIME_TYPE } from '../const'
import { render } from '../render'

describe('render', () => {
  it('can output svg', async () => {
    const svgString = await render({
      code: EXAMPLES.er.code,
      mimeType: SVG_MIME_TYPE,
    })
    expect(svgString.length).toBeGreaterThan(20)
  })

  it('can output jpeg', async () => {
    const buf = await render({
      code: EXAMPLES.er.code,
      mimeType: 'image/jpeg',
    })
    expect(buf.constructor).toBe(Buffer)
    expect(buf.length).toBeGreaterThan(0)
  })

  it('pintoraConfig should not alter global config', async () => {
    const oldTheme = pintoraStandalone.configApi.getConfig().themeConfig.theme
    const oldUseMaxWidth = pintoraStandalone.configApi.getConfig().core?.useMaxWidth
    await render({
      code: EXAMPLES.er.code,
      width: 1000,
      pintoraConfig: {
        themeConfig: {
          theme: 'light',
        },
      },
    })

    const config = pintoraStandalone.configApi.getConfig()
    expect(config.themeConfig.theme).toBe(oldTheme)
    expect(config.core.useMaxWidth).toBe(oldUseMaxWidth)
  })

  it('should cleanup global pollution after render', async () => {
    const fakeDom = {}
    ;(globalThis as any).document = fakeDom

    await render({
      code: EXAMPLES.er.code,
      mimeType: SVG_MIME_TYPE,
    })
    expect(global.window).toBeUndefined()
    expect((globalThis as any).document).toBe(fakeDom) // should not mess with existing globals

    delete (globalThis as any).document
  })
})
