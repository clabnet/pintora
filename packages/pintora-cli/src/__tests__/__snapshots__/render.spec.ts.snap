// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render can output svg 1`] = `"<svg width=\\"477\\" height=\\"409.0908203125\\" xmlns=\\"http://www.w3.org/2000/svg\\"><defs id=\\"defs_1\\"></defs><g opacity=\\"1\\" transform=\\"matrix(1,0,0,1,15,15)\\"><g opacity=\\"1\\" class=\\"er__relations\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 187.33333333333334 50L 99.5 100L 99.5 150\\" stroke=\\"#3b4044\\" stroke-linejoin=\\"round\\" fill=\\"none\\"></path><path opacity=\\"0.85\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 110.81892352164402,69.91388218644552 l 48,0 l 0,21.9609375 l-48 0 z\\" fill=\\"#f8f8f2\\" id=\\"#rel1\\" stroke=\\"none\\" class=\\"label-bg\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"134.81892352164402\\" y=\\"79.89435093644552\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" id=\\"rel1\\" fill=\\"#3b4044\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__relationship-label\\">places</text><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M9,-9 L9,9 M15,-9 L15,9\\" stroke=\\"#3b4044\\" id=\\"CUSTOMERplacesORDER-start\\" fill=\\"none\\" class=\\"er-marker er-marker--only-one\\" transform=\\"matrix(-0.869053840637207,0.494717538356781,-0.494717538356781,-0.869053840637207,187.3333282470703,50)\\"></path><g opacity=\\"1\\" class=\\"er-marker er-marker--zero-or-more\\" transform=\\"matrix(6.123234262925839e-17,-1,1,6.123234262925839e-17,99.5,150)\\"><circle opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" cx=\\"28\\" cy=\\"0\\" r=\\"6\\" stroke=\\"#3b4044\\" id=\\"CUSTOMERplacesORDER-end\\" fill=\\"#fff\\"></circle><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" stroke=\\"#3b4044\\" id=\\"CUSTOMERplacesORDER-end\\" d=\\"M-18,0 Q 0,18 18,0 Q 0,-18 -18,0\\" fill=\\"none\\"></path></g><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 99.5 229.0908203125L 99.5 279.0908203125L 99.5 329.0908203125\\" stroke=\\"#3b4044\\" stroke-linejoin=\\"round\\" fill=\\"none\\"></path><path opacity=\\"0.85\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 69.5,260.4296875 l 60,0 l 0,19.322265625 l-60 0 z\\" fill=\\"#f8f8f2\\" id=\\"#rel2\\" stroke=\\"none\\" class=\\"label-bg\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"99.5\\" y=\\"269.0908203125\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" id=\\"rel2\\" fill=\\"#3b4044\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__relationship-label\\">contains</text><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M9,-9 L9,9 M15,-9 L15,9\\" stroke=\\"#3b4044\\" id=\\"ORDERcontainsLINE-ITEM-start\\" fill=\\"none\\" class=\\"er-marker er-marker--only-one\\" transform=\\"matrix(6.123234262925839e-17,1,-1,6.123234262925839e-17,99.5,229.0908203125)\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M-18,0 Q 0,18 18,0 Q 0,-18 -18,0 M24,-9 L24,9\\" stroke=\\"#3b4044\\" id=\\"ORDERcontainsLINE-ITEM-end\\" fill=\\"none\\" class=\\"er-marker er-marker--one-or-more\\" transform=\\"matrix(6.123234262925839e-17,-1,1,6.123234262925839e-17,99.5,329.0908203125)\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 275.1666666666667 50L 363 100L 363 164.54541015625\\" stroke=\\"#3b4044\\" stroke-linejoin=\\"round\\" stroke-dasharray=\\"4,4\\" fill=\\"none\\"></path><path opacity=\\"0.85\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 314.23737420505273,75.37621441345506 l 37,0 l 0,16.79296875 l-37 0 z\\" fill=\\"#f8f8f2\\" id=\\"#rel3\\" stroke=\\"none\\" class=\\"label-bg\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"332.73737420505273\\" y=\\"82.77269878845506\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" id=\\"rel3\\" fill=\\"#3b4044\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__relationship-label\\">uses</text><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M-18,0 Q 0,18 18,0 Q 0,-18 -18,0 M24,-9 L24,9\\" stroke=\\"#3b4044\\" id=\\"CUSTOMERusesDELIVERY-ADDRESS-start\\" fill=\\"none\\" class=\\"er-marker er-marker--one-or-more\\" transform=\\"matrix(0.869053840637207,0.494717538356781,-0.494717538356781,0.869053840637207,275.1666564941406,50)\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M-18,0 Q 0,18 18,0 Q 0,-18 -18,0 M24,-9 L24,9\\" stroke=\\"#3b4044\\" id=\\"CUSTOMERusesDELIVERY-ADDRESS-end\\" fill=\\"none\\" class=\\"er-marker er-marker--one-or-more\\" transform=\\"matrix(6.123234262925839e-17,-1,1,6.123234262925839e-17,363,164.54541015625)\\"></path></g><g opacity=\\"1\\" class=\\"er__entity\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 178.5,0 l 105,0 a 2,2,0,0,1,2,2 l 0,46 a 2,2,0,0,1,-2,2 l -105,0 a 2,2,0,0,1,-2,-2 l 0,-46 a 2,2,0,0,1,2,-2 z\\" fill=\\"#fdb05e\\" stroke=\\"#3b4044\\" class=\\"er__entity-box\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"176.5\\" y=\\"0\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"bold\\" font-variant=\\"normal\\" fill=\\"#3b4044\\" width=\\"79\\" height=\\"10.609375\\" id=\\"entity-CUSTOMER\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,54.5,25)\\">CUSTOMER</text><g opacity=\\"1\\"></g></g><g opacity=\\"1\\" class=\\"er__entity\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 1.5,149.45458984375 l 195,0 a 2,2,0,0,1,2,2 l 0,75.0908203125 a 2,2,0,0,1,-2,2 l -195,0 a 2,2,0,0,1,-2,-2 l 0,-75.0908203125 a 2,2,0,0,1,2,-2 z\\" fill=\\"#fdb05e\\" stroke=\\"#3b4044\\" class=\\"er__entity-box\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"bold\\" font-variant=\\"normal\\" fill=\\"#3b4044\\" width=\\"50\\" height=\\"10.609375\\" id=\\"entity-ORDER\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,99.5,12.8046875)\\">ORDER</text><g opacity=\\"1\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M -0.5,175.06396484375 l 33,0 l 0,53.4814453125 l-33 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 32.5,200.40673828125 l 50,0 l 0,28.138671875 l-50 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 82.5,200.40673828125 l 116,0 l 0,28.138671875 l-116 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 32.5,175.06396484375 l 50,0 l 0,25.3427734375 l-50 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 82.5,175.06396484375 l 116,0 l 0,25.3427734375 l-116 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M -0.5,175.06396484375 l 33,0 l 0,25.3427734375 l-33 0 z\\" fill=\\"#fffbf9\\" stroke=\\"#3b4044\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"left\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"11.9\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" width=\\"18\\" height=\\"10.0419921875\\" fill=\\"#3b4044\\" id=\\"entity-ORDER-attr-1-key\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,7.5,38.28076171875)\\">PK</text><text alignment-baseline=\\"central\\" text-anchor=\\"left\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"11.9\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" width=\\"15\\" height=\\"10.158203125\\" fill=\\"#3b4044\\" id=\\"entity-ORDER-attr-1-type\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,40.5,38.28076171875)\\">int</text><text alignment-baseline=\\"central\\" text-anchor=\\"left\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"11.9\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" width=\\"85\\" height=\\"10.3427734375\\" fill=\\"#3b4044\\" id=\\"entity-ORDER-attr-1-name\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,90.5,38.28076171875)\\">orderNumber</text><text alignment-baseline=\\"central\\" text-anchor=\\"left\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"11.9\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" width=\\"35\\" height=\\"13.138671875\\" fill=\\"#3b4044\\" id=\\"entity-ORDER-attr-2-type\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,40.5,65.021484375)\\">string</text><text alignment-baseline=\\"central\\" text-anchor=\\"left\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"-0.5\\" y=\\"149.45458984375\\" font-size=\\"11.9\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"normal\\" font-variant=\\"normal\\" width=\\"101\\" height=\\"13.0771484375\\" fill=\\"#3b4044\\" id=\\"entity-ORDER-attr-2-name\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,90.5,65.021484375)\\">deliveryAddress</text></g></g><g opacity=\\"1\\" class=\\"er__entity\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 51,329 l 96,0 a 2,2,0,0,1,2,2 l 0,46 a 2,2,0,0,1,-2,2 l -96,0 a 2,2,0,0,1,-2,-2 l 0,-46 a 2,2,0,0,1,2,-2 z\\" fill=\\"#fdb05e\\" stroke=\\"#3b4044\\" class=\\"er__entity-box\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"49\\" y=\\"329\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"bold\\" font-variant=\\"normal\\" fill=\\"#3b4044\\" width=\\"70\\" height=\\"10.0419921875\\" id=\\"entity-LINE-ITEM\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,50,25)\\">LINE-ITEM</text><g opacity=\\"1\\"></g></g><g opacity=\\"1\\" class=\\"er__entity\\"><path opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" d=\\"M 281,164 l 164,0 a 2,2,0,0,1,2,2 l 0,46 a 2,2,0,0,1,-2,2 l -164,0 a 2,2,0,0,1,-2,-2 l 0,-46 a 2,2,0,0,1,2,-2 z\\" fill=\\"#fdb05e\\" stroke=\\"#3b4044\\" class=\\"er__entity-box\\"></path><text alignment-baseline=\\"central\\" text-anchor=\\"middle\\" opacity=\\"1\\" stroke-width=\\"1\\" stroke-opacity=\\"1\\" fill-opacity=\\"1\\" x=\\"279\\" y=\\"164\\" font-size=\\"14\\" font-family=\\"sans-serif\\" font-style=\\"normal\\" font-weight=\\"bold\\" font-variant=\\"normal\\" fill=\\"#3b4044\\" width=\\"138\\" height=\\"10.609375\\" id=\\"entity-DELIVERY-ADDRESS\\" paint-order=\\"stroke\\" style=\\"stroke-linecap:butt; stroke-linejoin:miter;\\" stroke=\\"none\\" class=\\"er__entity-label\\" transform=\\"matrix(1,0,0,1,84,25)\\">DELIVERY-ADDRESS</text><g opacity=\\"1\\"></g></g></g></svg>"`;
